wfe/wfe.go

```diff
@@ -1115,6 +1120,19 @@ func (wfe *WebFrontEndImpl) Registration(ctx context.Context, logEvent *requestE
 		return
 	}
 
+	// People *will* POST their full registrations to this endpoint, including
+	// the 'valid' status, to avoid always failing out when that happens only
+	// attempt to deactivate if the provided status different from their current
+	// status.
+	if wfe.AllowAccountDeactivation && (update.Status != currReg.Status) {
+		if update.Status != core.StatusDeactivated {
+			wfe.sendError(response, logEvent, probs.Malformed("Invalid value provided for status field"), nil)
+			return
+		}
+		wfe.deactivateRegistration(ctx, currReg, response, request, logEvent)
+		return
```

Comment:
If someone POSTs with, say, a deactivation and a contact update, we will deactivate without updating the contact. This is reasonable but doesn't exactly match the implied REST semantics. So we should document it clearly in comments, and test for it. The comments should also indicate specifically that the deactivation stanza must be first in the list of checks.